test/integration/uds_integration_test.h

```diff
@@ -57,14 +58,20 @@ class UdsListenerIntegrationTest : public HttpIntegrationTest,
   void initialize() override;
 
   std::string getAdminSocketName() {
+    const testing::TestInfo* testInfo = testing::UnitTest::GetInstance()->current_test_info();
     return abstract_namespace_
-               ? "@/envoy-tests/uds-integration-test/admin"
-               : TestEnvironment::unixDomainSocketPath("uds-integration-test.admin.sock");
+               ? fmt::format("@/envoy-tests/{}/{}/admin", testInfo->test_case_name(),
```

Comment:
I'm worried about the case where multiple top-level builds are running concurrently on a box. These abstract-namespace UDS paths are global to the box, which is great for their intended use case, and difficult for making these tests work. @htuch is there any precedent anywhere else in the tests for a collision type like this? Should we put a random number in the path? Or maybe call mktmp and use the pathname it returns?

Sorry to be so picky about this; just trying to avoid flaky tests.

Reply:
Note that the existing UDS integration test is using abstract sockets for its admin address.

Reply:
Yes, this is an issue, see what was done recently in https://github.com/envoyproxy/envoy/blob/b6b15645e1bc740d5280dee4561ff3a3cb0cc5fa/test/exe/main_common_test.cc#L47. Can we add a `unixDomainSocketAbstract()` util that does the same as `TestEnvironment::unixDomainSocketPath` for abstract sockets? We could have a per-instance global for the entropy.

Reply:
OK, I changed this to take the output of `TestEnvironment::unixDomainSocketPath` and prepend `"@"` for abstract sockets. This should be sufficiently random, since the file paths are already sharing a system-wide temp dir.